packages/clerk-js/src/ui/elements/Form.tsx

```diff
@@ -214,6 +229,85 @@ const Checkbox = (
   );
 };
 
+const LegalCheckboxLabel = (props: { termsUrl?: string; privacyPolicyUrl?: string }) => {
+  const { t } = useLocalizations();
+  return (
+    <Text
+      variant='body'
+      as='span'
+    >
+      {t(localizationKeys('signUp.legalConsent.checkbox.label__prefixText'))}
+      {props.termsUrl && (
+        <>
+          {' '}
+          <Link
+            localizationKey={localizationKeys('signUp.legalConsent.checkbox.label__termsOfServiceText')}
+            href={props.termsUrl}
+            sx={{
+              textDecoration: 'underline',
+            }}
+            isExternal
+          />
+        </>
+      )}
+
+      {props.termsUrl && props.privacyPolicyUrl && (
+        <> {t(localizationKeys('signUp.legalConsent.checkbox.label__conjunctionText'))} </>
+      )}
+
+      {props.privacyPolicyUrl && (
+        <>
+          {' '}
+          <Link
+            localizationKey={localizationKeys('signUp.legalConsent.checkbox.label__privacyPolicyText')}
+            href={props.termsUrl}
+            sx={{
+              textDecoration: 'underline',
+              display: 'inline-block',
+            }}
+            isExternal
+          />
+        </>
+      )}
+    </Text>
+  );
+};
+
+const LegalCheckbox = (
+  props: CommonFieldRootProps & {
+    description?: string | LocalizationKey;
+  },
+) => {
+  const { displayConfig } = useEnvironment();
+  const { parsedLayout } = useAppearance();
+
+  const termsLink = parsedLayout.termsPageUrl || displayConfig.termsUrl;
+  const privacyPolicy = parsedLayout.privacyPageUrl || displayConfig.privacyPolicyUrl;
+
+  return (
+    <Field.Root {...props}>
+      <Flex
+        align='center'
+        justify='center'
+      >
+        <Field.CheckboxIndicator />
```

Comment:
Will want to ensure there is a focus style for the checkbox, currently missing due to the usage for `focusRing={false}` on the `CheckboxIndicator`. Turning that to true renders a ring but its much to big in my opinion. Are we able to add a focus ring for [focus visible](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible)?

Reply:
Nice suggestion, yeah we can add that!